### PR TITLE
Add a move constructor for VectorMemory::Pointer.

### PR DESCRIPTION
**Pull request by drwells**
This lets us write things like
```cpp
std::vector<typename VectorMemory<VectorType>::Pointer> vec;
```

Also part of #5621.

### CODE REVIEW

File: include/deal.II/lac/vector_memory.h [hunk 1]
```diff
@@ -196,6 +196,12 @@ class VectorMemory : public Subscriptor
      */
     Pointer() = default;
 
+    /**
+     * Move constructor. This operator steals the pointer to the vector
+     * represented by this object from the one give as argument.
```
Comment (masterleinad) [inline include/deal.II/lac/vector_memory.h]: `give`->`given` (and below for the move operator).

**Comment by drwells** [comment]
Good point. I reworded both.

**Comment by bangerth** [comment]
OK if you could add a short test along the lines of the one given in the original report.

**Comment by drwells** [comment]
I added a test based on the original issue. This won't work, though, without #5628, so we should merge that first.

**Comment by bangerth** [comment]
Oh, I completely misunderstood the original test -- I thought the pointers were to go into a std::vector, not that there should be a `GrowingVectorMemory<std::vector<...>>`. The former makes eminent sense to me, the latter not so much. (Though it also doesn't hurt.)

Either way, there is no harm in supporting it. #5628 is independently useful.

**Comment by drwells** [comment]
@bangerth My understanding of #5621 is that we did not support either of those; #5628 lets us do `GrowingVectorMemory<std::vector<T>>` and this PR lets us do `std::vector<VectorMemory::Pointer<T>>`.

**Pull request merged**